spec/diagnose_spec.rb

```diff
@@ -23,114 +23,142 @@
     @runner.run
   end
 
-  it "prints the diagnose header" do
-    expect_output([
-      /AppSignal diagnose/,
-      /================================================================================/,
-      /Use this information to debug your configuration./,
-      /More information is available on the documentation site./,
-      %r{https://docs.appsignal.com/},
-      /Send this output to support@appsignal.com if you need help./,
-      /================================================================================/
-    ])
+  it "prints all sections in the correct order" do
+    section_keys =
+      case @runner.type
+      when :ruby
+        [
+          :header,
+          :library,
+          :installation,
+          :host,
+          :agent,
+          :config,
+          :validation,
+          :paths,
+          :send_report
+        ]
+      when :nodejs
+        [
+          :header,
+          :library,
+          :installation,
+          :host,
+          # TODO: Add agent section for Node.js
+          :config,
+          :validation,
+          :paths,
+          :send_report
+        ]
+      end
+    expect(@runner.output.sections.keys).to eq(section_keys)
   end
 
-  it "prints a newline" do
-    expect_newline
+  it "prints no 'other' section" do
+    # Any output that couldn't be categorized are part of the "other" category.
+    # It should not exist.
+    expect(section(:other)).to eq("") if @runner.output.sections.key?(:other)
```

Comment:
Shouldn't we just `expect(@runner.output.sections).not_to include(:other)`? Is there a case where it is expected to exist as an empty string?

Reply:
The key may not exist. I don't want to only check that key presence because the failure output will be unhelpful I think. It's a bit of an ugly solution, but also only temporary solution while we improve these tests.

Given the assertion currently committed:

```
     Failure/Error: expect(section(:other)).to eq("") if @runner.output.sections.key?(:other)

       expected: ""
            got: "AppSignal diagnose\n================================================================================...n  Send diagnostics report to AppSignal? (Y/n):   Not sending diagnostics information to AppSignal."

# But it then prints the output of the "other" section as not matching with the empty string so you can see which lines were printed in that section
```

This assertion gives the following output, trimming the contents of the `Output` instance which makes it impossible to tell what are the contents of the "other" section.

```ruby
expect(@runner.output.sections).to_not have_key(:other)
```

```
     Failure/Error: expect(@runner.output.sections).to_not have_key(:other)
       expected `{:config=>["Configuration", "  Environment: 'test'", "  debug: false", "  log: 'file'", "  endpoint: ...n to AppSignal."], :validation=>["Validation", "  Validating Push API key: \e[31minvalid\e[0m", ""]}.has_key?(:other)` to be falsey, got true
```

Reply:
I see! Makes sense to me. 👍